app/operations/story_operations.rb

```diff
@@ -144,11 +144,31 @@ def create_activity(story, current_user:)
 
   class UpdateAll < BaseOperations::UpdateAll; end
 
-  class Destroy < BaseOperations::Destroy
-    include LegacyPusherNotification
+  class Destroy
+    include Dry::Monads[:result, :do]
+
+    def call(params)
+      story = yield delete_story(params[:story])
+      yield notify_changes(story)
+      yield create_activity(story, current_user: params[:current_user])
```

Comment:
What do you think about using the method `fetch` instead of `[]`? This way, if the key `story` or `current_user` is missing it will throw an error: https://ruby-doc.org/core-3.0.1/Hash.html#method-i-fetch

```suggestion
      story = yield delete_story(params.fetch(:story))
      yield notify_changes(story)
      yield create_activity(story, current_user: params.fetch(:current_user))
```

We could do the same to the other operations. Alternatively we could use named params instead of a hash.

Reply:
I agree it's better to go with named params!